### ISSUE TITLE
Recognize foot as an xterm-compatible terminal emulator

### ISSUE THREAD

**Issue by mc-butler** [MidnightCommander/mc#4251]
> [!IMPORTANT]
> This issue was migrated from Trac:
>
> | **Origin** | https://midnight-commander.org/ticket/4251 |
> | :------------- | :------------------------------------------------------------------------------------------------ |
> | **Reporter** | `gszymaszek` (@gszy) |

<p>
foot is a "fast, lightweight and minimalistic Wayland terminal emulator" (see <a class="ext-link" href="https://codeberg.org/dnkl/foot">foot on Codeberg</a>). It is recommended by the <a class="ext-link" href="https://packages.debian.org/bullseye/sway">Debian sway package</a>. Add its default <tt>TERM</tt> value to the recognized ones. This enables mc to change the terminal window title and handle mouse events.<br />
</p>

> [!NOTE]
> Original attachments:
>
> * [0001-Add-foot-to-the-known-terminals.patch](https://github.com/MidnightCommander/trac-archive/blob/master/attachments/ticket/4251/0001-Add-foot-to-the-known-terminals.patch) ([raw](https://github.com/MidnightCommander/trac-archive/raw/master/attachments/ticket/4251/0001-Add-foot-to-the-known-terminals.patch)) by `gszymaszek` (@gszy) on <span title="2021-05-29T16:31:36+00:00">`May 29, 2021 at 16:31 UTC`</span>

**Comment by unoukujou** [comment]
> > Changed by `andrew_b` ([@aborodin](https://github.com/aborodin)) on `May 30, 2021 at 10:22 UTC` _(comment 5)_
> 
> * **Status** changed from _testing_ to _closed_

Hello.

I'm not sure if this is fully completed.

I think there needs to be an entry for foot in this file too:

misc/mc.lib
https://github.com/MidnightCommander/mc/blob/dc2dbc804e44e5bfd9643cb53dcd2bc5d57145f7/misc/mc.lib

I don't know how to explain it, but... When I use foot terminal and I open midnight commander, shift+F4 becomes shift+F6, Shift+F6 becomes Shift+F8... It's like it's adding an extra 2 to every shift + F key.

This is when: 
$TERM = foot

If I open foot terminal and set $TERM=alacritty for example, now all of a sudden Midnight Commander works properly. At first I thought it could be an issue with foot terminal, but after seeing this behavior, I realized that it's something with Midnight Commander. Because midnight commander works fine in foot terminal as long as I set $TERM=alacritty

I don't know if the problem is with this file (misc/mc.lib) but after doing a search through the source, that looks like it could be the reason. But again, I'm not sure...
